pylint/extensions/_check_docs_utils.py

```diff
@@ -164,10 +164,17 @@ def possible_exc_types(node):
                     and isinstance(val, (astroid.Instance, nodes.ClassDef))
                     and utils.inherit_from_std_ex(val)
                 ):
-                    excs.append(val.name)
+                    excs.append(val)
 
     try:
-        return {exc for exc in excs if not utils.node_ignores_exception(node, exc)}
+        final_exceptions = set()
+        for exc in excs:
+            if utils.node_ignores_exception(node, exc.name):
+                continue
+            if not isinstance(exc, nodes.ClassDef):
```

Comment:
Why can `exc` not be a `ClassDef` at this point? The coverage report shows that this line is covered by the tests so it seems necessary but I wonder if we shouldn't "fix" this before iterating over `excs`.

Reply:
I guess there is one corner case. I can't remember which test case but this line is needed. Also this follows the old logic. To be honest I don't feel comfortable to dig too deep, since I'm now expert in this code